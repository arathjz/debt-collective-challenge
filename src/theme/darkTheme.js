module.exports = {
  main: '#162447',
  secondary: '#1f4068',
  alternary: '#1b1b2f',
  textColor: '#f0f0f0',
};
