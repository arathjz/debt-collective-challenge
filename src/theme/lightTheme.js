module.exports = {
  main: '#f0f0f0',
  secondary: '#1b6ca8',
  alternary: '#ffffff',
  textColor: '#162447',
};
