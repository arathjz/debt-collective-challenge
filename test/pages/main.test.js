/* eslint-disable react/jsx-pascal-case */
import React from 'react';
import { useStaticQuery } from 'gatsby';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { ThemeProvider } from 'styled-components';

import Main from '../../src/pages';
import theme from '../../src/theme/darkTheme';

jest.mock('../../src/hooks/useOrganization', () => {
  return () => ({
    github: {
      organization: {
        name: 'The debt collective',
        avatarUrl: 'https://images.com/image.svg',
        login: 'debtcollective',
        url: 'https://github.com/debtcollective',
      },
    },
  });
});

describe('Main', () => {
  beforeEach(() => {
    useStaticQuery.mockImplementation(() => ({
      placeholderImage: {
        childImageSharp: {
          fixed: {
            base64: 'image',
            width: 50,
            height: 50,
            src: '/static/image.png',
            srcSet: '/static/image.png',
          },
        },
      },
    }));
  });

  test('should render correctly', () => {
    const container = render(
      <ThemeProvider theme={theme}>
        <Main />
      </ThemeProvider>,
    );

    expect(container).toMatchSnapshot();
  });
});
