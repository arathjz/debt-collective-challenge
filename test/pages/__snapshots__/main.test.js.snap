// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main should render correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-AxhCb BIYNM"
      >
        <header
          class="sc-AxiKw bQuuop"
        >
          <div
            class="content"
          >
            <a
              class="sc-AxirZ cLMVqT"
              href="/"
            >
              <div
                class=" gatsby-image-wrapper"
                style="position: relative; overflow: hidden; display: inline-block; width: 50px; height: 50px;"
              >
                <img
                  alt=""
                  aria-hidden="true"
                  src="image"
                  style="position: absolute; top: 0px; left: 0px; width: 100%; height: 100%; object-fit: cover; object-position: center; opacity: 1; transition-delay: 500ms;"
                />
                <picture>
                  <source
                    srcset="/static/image.png"
                  />
                  <img
                    alt=""
                    height="50"
                    loading="lazy"
                    src="/static/image.png"
                    srcset="/static/image.png"
                    style="position: absolute; top: 0px; left: 0px; width: 100%; height: 100%; object-fit: cover; object-position: center; opacity: 0; transition: opacity 500ms;"
                    width="50"
                  />
                </picture>
                <noscript>
                  &lt;picture&gt;&lt;source srcset="/static/image.png" /&gt;&lt;img loading="lazy" width="50" height="50" srcset="/static/image.png" src="/static/image.png" alt="" style="position:absolute;top:0;left:0;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center"/&gt;&lt;/picture&gt;
                </noscript>
              </div>
              <h1 />
            </a>
            <button
              aria-label="Left Align"
              class="sc-AxjAm esyKCT"
              type="button"
            >
              <img
                alt="toggle dark/light theme"
                src="test-file-stub"
              />
            </button>
          </div>
        </header>
        <div>
          <div
            class="sc-AxhUy eYAjWC"
          >
            <h2>
              Welcome!
            </h2>
            <a
              class="sc-AxgMl inxGsF"
              href="/repositories"
            >
              View repositories
            </a>
          </div>
        </div>
        <footer>
          <p>
            Icons powered by
             
            <a
              href="https://fontawesome.com/license"
              rel="noreferrer"
              target="_blank"
            >
              FontAwesome
            </a>
          </p>
          <p>
            Build with ♥ by
             
            <a
              href="https://github.com/arathjz"
              rel="noreferrer"
              target="_blank"
            >
              @arathjz
            </a>
             
          </p>
        </footer>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-AxhCb BIYNM"
    >
      <header
        class="sc-AxiKw bQuuop"
      >
        <div
          class="content"
        >
          <a
            class="sc-AxirZ cLMVqT"
            href="/"
          >
            <div
              class=" gatsby-image-wrapper"
              style="position: relative; overflow: hidden; display: inline-block; width: 50px; height: 50px;"
            >
              <img
                alt=""
                aria-hidden="true"
                src="image"
                style="position: absolute; top: 0px; left: 0px; width: 100%; height: 100%; object-fit: cover; object-position: center; opacity: 1; transition-delay: 500ms;"
              />
              <picture>
                <source
                  srcset="/static/image.png"
                />
                <img
                  alt=""
                  height="50"
                  loading="lazy"
                  src="/static/image.png"
                  srcset="/static/image.png"
                  style="position: absolute; top: 0px; left: 0px; width: 100%; height: 100%; object-fit: cover; object-position: center; opacity: 0; transition: opacity 500ms;"
                  width="50"
                />
              </picture>
              <noscript>
                &lt;picture&gt;&lt;source srcset="/static/image.png" /&gt;&lt;img loading="lazy" width="50" height="50" srcset="/static/image.png" src="/static/image.png" alt="" style="position:absolute;top:0;left:0;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center"/&gt;&lt;/picture&gt;
              </noscript>
            </div>
            <h1 />
          </a>
          <button
            aria-label="Left Align"
            class="sc-AxjAm esyKCT"
            type="button"
          >
            <img
              alt="toggle dark/light theme"
              src="test-file-stub"
            />
          </button>
        </div>
      </header>
      <div>
        <div
          class="sc-AxhUy eYAjWC"
        >
          <h2>
            Welcome!
          </h2>
          <a
            class="sc-AxgMl inxGsF"
            href="/repositories"
          >
            View repositories
          </a>
        </div>
      </div>
      <footer>
        <p>
          Icons powered by
           
          <a
            href="https://fontawesome.com/license"
            rel="noreferrer"
            target="_blank"
          >
            FontAwesome
          </a>
        </p>
        <p>
          Build with ♥ by
           
          <a
            href="https://github.com/arathjz"
            rel="noreferrer"
            target="_blank"
          >
            @arathjz
          </a>
           
        </p>
      </footer>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
